core/embed/rust/src/translations/flash.rs

```diff
@@ -1,28 +1,40 @@
+use spin::{RwLock, RwLockReadGuard};
+
 use crate::{error::Error, trezorhal::translations};
 
 use super::blob::Translations;
 
-static mut TRANSLATIONS_ON_FLASH: Option<Translations> = None;
+static TRANSLATIONS_ON_FLASH: RwLock<Option<Translations>> = RwLock::new(None);
 
+/// Erase translations blob from flash.
+///
+/// The blob must be deinitialized via `deinit()` before calling this function.
 pub fn erase() -> Result<(), Error> {
-    // SAFETY: Looking is safe (in a single threaded environment).
-    if unsafe { TRANSLATIONS_ON_FLASH.is_some() } {
-        return Err(value_error!("Translations blob already set"));
-    }
+    let blob = TRANSLATIONS_ON_FLASH.read();
```

Comment:
Conceptually, shouldn't a write-lock be taken here? Not that it really matters since there's the `is_some` check.

Reply:
I originally thought like, this is a write operation on flash, not on the translations object.

but if we look at the module as a whole, the lock is kind-of for "the whole API" so it would make more sense to enforce that, e.g., only one writer is allowed to write to flash at the same time 

(then again we don't have threads so 🤷‍♀️ )

Reply:
a6bec64b71c96129382a08e8a464405ecf9bab3a